src/Jenkinsfile.groovy

```diff
@@ -74,6 +75,7 @@ class Jenkinsfile {
     }
 
     public static build(Closure closure) {
+        original.ApplyJenkinsfileClosure(createParamClosure())
```

Comment:
I'm getting a little lost on how to support adding in the parameters closure. I think the way I have it set up now, it is adding the params before each stage, which is not what we want. 

If we want to add the parameters ONLY once (before the stages are even defined), I was thinking about setting up something like this

```
    public static void build(List<Stage> stages) {
        if (!declarative) {
            original.ApplyJenkinsfileClosure(createParamClosure())
            stages.each { Stage stage -> stage.build() }
        } else {
            if (pipelineTemplate == null) {
                this.pipelineTemplate = getPipelineTemplate(stages)
            }
            pipelineTemplate.call(createParamClosure())
            pipelineTemplate.call(stages)
        }
    }
```

However this is causing the BuildGraphTest to fail with `ava.lang.NullPointerException: Cannot invoke method ApplyJenkinsfileClosure() on null object`